jenkins/spark-premerge-build.sh

```diff
@@ -160,7 +160,17 @@ ci_2() {
     # Test a portion of cases for non-UTC time zone because of limited GPU resources.
     # Here testing: parquet scan, orc scan, csv scan, cast, TimeZoneAwareExpression, FromUTCTimestamp
     # Nightly CIs will cover all the cases.
-    TZ=Iran TEST='test_parquet_read_round_trip or test_read_round_trip or test_basic_csv_read or test_cast_string_ts_valid_format or test_unix_timestamp or test_from_utc_timestamp' ./integration_tests/run_pyspark_from_build.sh
+    #
+    # potential corner cases:
+    # Hebron and Newfoundland are chosen because they contain many zone transitions.
+    # Belize is chosen for containing the most intra-hour zone transitions.
+    # Punta_Arenas is chosen for containing the most intra-minute zone transitions.
+    # Morocco(Africa/Casablanca) is chosen for frequently changing transition rules between the release of Java 8 and Java 17
+    non_utc_time_zones=("Iran" "Asia/Hebron" "Canada/Newfoundland" "America/Belize" "America/Punta_Arenas" "Africa/Casablanca")
+    for tz in "${non_utc_time_zones[@]}"
+    do
+        TZ=$tz TEST='test_parquet_read_round_trip or test_read_round_trip or test_basic_csv_read or test_cast_string_ts_valid_format or test_unix_timestamp or test_from_utc_timestamp' ./integration_tests/run_pyspark_from_build.sh
```

Comment:
I personally would rather see us tag specific time zone tests that we want instead of having the list hard coded here. The number of tests we care about is likely to increase as we go forward with the time zone work and adding more here is not ideal.

Reply:
Hi @revans2, I extracted the test list of time zones as a common asset. Then, we only need to modify the single spot when changing the time zones of the integration test. It isn't easy to push down the time zone tag to the Python test files because we have to set the time zone as `extraJavaOpts`.